src/protocols/secure_channel/PASESession.h

```diff
@@ -256,19 +256,21 @@ class DLL_EXPORT PASESession : public Messaging::ExchangeDelegate, public Pairin
     CHIP_ERROR SetupSpake2p(uint32_t pbkdf2IterCount, const ByteSpan & salt);
 
     CHIP_ERROR SendPBKDFParamRequest();
-    CHIP_ERROR HandlePBKDFParamRequest(const System::PacketBufferHandle & msg);
+    CHIP_ERROR HandlePBKDFParamRequest(System::PacketBufferHandle && msg);
 
-    CHIP_ERROR SendPBKDFParamResponse();
-    CHIP_ERROR HandlePBKDFParamResponse(const System::PacketBufferHandle & msg);
+    CHIP_ERROR SendPBKDFParamResponse(ByteSpan initiatorRandom, bool initiatorHasPBKDFParams);
+    CHIP_ERROR HandlePBKDFParamResponse(System::PacketBufferHandle && msg);
 
     CHIP_ERROR SendMsg1();
 
-    CHIP_ERROR HandleMsg1_and_SendMsg2(const System::PacketBufferHandle & msg);
-    CHIP_ERROR HandleMsg2_and_SendMsg3(const System::PacketBufferHandle & msg);
-    CHIP_ERROR HandleMsg3(const System::PacketBufferHandle & msg);
+    CHIP_ERROR HandleMsg1_and_SendMsg2(System::PacketBufferHandle && msg);
+    CHIP_ERROR HandleMsg2_and_SendMsg3(System::PacketBufferHandle && msg);
+    CHIP_ERROR HandleMsg3(System::PacketBufferHandle && msg);
 
-    void SendErrorMsg(Spake2pErrorType errorCode);
-    CHIP_ERROR HandleErrorMsg(const System::PacketBufferHandle & msg);
+    void SendStatusReport(uint16_t protocolCode);
+    CHIP_ERROR HandleStatusReport(System::PacketBufferHandle && msg);
+
+    constexpr size_t EstimateTLVStructOverhead(size_t dataLen, size_t nFields) { return dataLen + (sizeof(uint64_t) * nFields); }
```

Comment:
We should not cut'n'paste this around . This should be refactored to be part of TLV APIs.

Reply:
Added a TODO to move it to CHIPTLV.